src/new-storage/DeleteExpiredCmd.js

```diff
@@ -0,0 +1,139 @@
+const cassandra = require('cassandra-driver')
+const fetch = require('node-fetch')
+const pLimit = require('p-limit')
+
+const validateConfig = require('../helpers/validateConfig')
+
+class DeleteExpiredCmd {
+    constructor(config) {
+        validateConfig(config)
+
+        this.baseUrl = config.streamrUrl
+
+        const authProvider = new cassandra.auth.PlainTextAuthProvider(config.cassandraNew.username, config.cassandraNew.password)
+        this.cassandraClient = new cassandra.Client({
+            contactPoints: [...config.cassandraNew.hosts],
+            localDataCenter: config.cassandraNew.datacenter,
+            keyspace: config.cassandraNew.keyspace,
+            authProvider,
+            pooling: {
+                maxRequestsPerConnection: 32768
+            }
+        })
+
+        // used for limited concurrency
+        this.limit = pLimit(5)
+    }
+
+    async _getStreams() {
+        const result = []
+
+        const query = 'SELECT DISTINCT stream_id, partition FROM bucket'
+        const resultSet = await this.cassandraClient.execute(query)
+
+        if (resultSet.rows.length) {
+            resultSet.rows.forEach((row) => {
+                result.push({
+                    streamId: row.stream_id,
+                    partition: row.partition
+                })
+            })
+        }
+
+        return result
+    }
+
+    async _fetchStreamsInfo(streams) {
+        const tasks = streams.map((stream) => {
+            return this.limit(async () => {
+                const url = `${this.baseUrl}/api/v1/streams/${stream.streamId}/validation`
+                return fetch(url).then((res) => res.json()).then((json) => {
+                    return {
+                        streamId: stream.streamId,
+                        partition: stream.partition,
+                        storageDays: parseInt(json.storageDays)
+                    }
+                }).catch((e) => console.error(e))
+            })
+        })
+
+        return Promise.all(tasks)
+    }
+
+    async _deleteExpired(streamsInfo) {
+        const tasks = streamsInfo.map((stream) => {
+            const { bucketId, dateCreate, streamId, partition } = stream
+            const queries = [
+                {
+                    query: 'DELETE FROM bucket WHERE stream_id = ? AND partition = ? AND date_create = ?',
+                    params: [streamId, partition, dateCreate]
+                },
+                {
+                    query: 'DELETE FROM stream_data WHERE stream_id = ? AND partition = ? AND bucket_id = ?',
```

Comment:
Did we discuss should we delete only buckets or buckets+data? I'm a  bit afraid of tombstones if we delete data.

Reply:
We can discuss it on network call or today. 
And we can run in addition `nodetool compact <keyspace> <table>`